storage.go

```diff
@@ -31,20 +33,57 @@ func (s *Storage) delete(ctx context.Context, path string, opt *pairStorageDelet
 	}
 	return
 }
-func (s *Storage) listDir(ctx context.Context, dir string, opt *pairStorageListDir) (err error) {
+
+func (s *Storage) list(ctx context.Context, path string, opt pairStorageList) (oi *ObjectIterator, err error) {
+	input := &objectPageStatus{
+		limit:  200,
+		prefix: s.getAbsPath(path),
+	}
+	if opt.HasContinuationToken {
+		input.marker = opt.ContinuationToken
+	}
+
+	var nextFn NextObjectFunc
+
+	switch {
+	case opt.ListMode.IsDir():
+		input.delimiter = "/"
+		nextFn = s.nextObjectPageByDir
+	case opt.ListMode.IsPrefix():
+		nextFn = s.nextObjectPageByPrefix
+	default:
+		return nil, fmt.Errorf("invalid list mode")
+	}
+
+	return NewObjectIterator(ctx, nextFn, input), nil
+}
+
+func (s *Storage) metadata(ctx context.Context, opt pairStorageMetadata) (meta *StorageMeta, err error) {
+	meta = NewStorageMeta()
+	meta.Name = s.name
+	meta.WorkDir = s.workDir
+	return meta, nil
+}
+
+func (s *Storage) nextObjectPageByDir(ctx context.Context, page *ObjectPage) (err error) {
+	input := page.Status.(*objectPageStatus)
+
+	header := make(map[string]string)
+	header["X-List-Limit"] = strconv.Itoa(input.limit)
```

Comment:
Make sure the type in `objectPageStatus` is the same with what we needed to use, so we can reduce the extra convertion.

As this place, let's use `limit` as string and use `iter` instead of `marker`

Reply:
OK, I thought that the fields' name are all the same

Reply:
`limit` is also used as cap of `FileInfo` channel, so I think it is better to keep it `int`.

Reply:
FileInfo's channel cap is another thing. In fact, this channel should always be clear because we will fetch them to our own array.

Reply:
Changing the cap into 1 is enough to avoid data block